data/transactions/logic/eval.go

```diff
@@ -584,7 +587,31 @@ func (cx *evalContext) step() {
 		cx.err = fmt.Errorf("pc=%3d dynamic cost budget of %d exceeded, executing %s", cx.pc, cx.budget(), spec.Name)
 		return
 	}
+
+	preheight := len(cx.stack)
 	spec.op(cx)
+
+	if cx.err == nil {
```

Comment:
I don't think so.  It only checks things that would have previously caused an error.  But now it does it in a central place, so that future opcodes won't have to do it themselves.  In particular, in the past only concat could create a 4k slice, so only concat checked its output.  Now we check here, in case we have introduced a new way to create a 4k slice (I don't think we have, since byteslice math is input limited, but this seems like a useful thing to centralize, especially as we might add other ways to hit 4k)